src/schema/sources.ts

```diff
@@ -1481,6 +1539,67 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         true,
       );
     },
+    sourceAwards: async (
+      _,
+      args: GQLCommentAwardArgs,
+      ctx: Context,
+      info,
+    ): Promise<Connection<GQLUserAward>> => {
+      await ensureSourcePermissions(ctx, args.id);
+      await ctx.con.getRepository(Source).findOneOrFail({
```

Comment:
Since we already do `ensureSourcePermissions`, we don't really need the `findOneOrFail` here as long as we don't actually use it for anything. The `ensureSourcePermissions` already does a `findOneByOrFail`